### ISSUE TITLE
Compiling on ArchLinux

### ISSUE THREAD

**Issue by tkh23** [MyHush/hush#61]
Attempting to build on ArchLinux using slightly modified zcash PKGBUILD (attached). Downloads sources and begins compiling, after about 10-15 minutes, following error:
make[2]: *** No rule to make target 'gtest/test_wallet_zkeys.cpp', needed by 'gtest/zcash_gtest-test_wallet_zkeys.o'.  Stop.
[PKGBUILD.txt](https://github.com/MyHush/hush/files/1415582/PKGBUILD.txt)

**Comment by leto** [comment]
@tkh23 please post the full output in a Github gist or something, we need more details to debug

**Comment by leto** [comment]
@tkh23 following the issue template makes things easier for everybody 😸

**Comment by madbuda** [comment]
If you follow instructions from install.md it builds fine on arch. Unless you have gcc7

**Comment by tkh23** [comment]
Thank you both. I will check the instructions on install.md and see if I can adjust the PKGBUILD accordingly. Then I'll report back, and if issues persist, I will use the issue template (thanks for pointing out) and include full output.

**Comment by tkh23** [comment]
I've installed gcc6 (alongside latest gcc7, ie. /usr/bin/gcc-6), but I can't figure out how to tell the system to use it. I've been googling for an hour and I'm getting sick of it. Yeah, I could temporarily uninstall or backup gcc7 and ln -s, but that's not a solution. Any help would be appreciated.
Also, why doesn't hush support gcc7 anyways?

**Comment by madbuda** [comment]
https://github.com/MyHush/hush/blob/master/zcutil/build.sh#L103

I miss typed, we are still using gcc 5, I'm assuming since you are using pkgbuild it is suppressing the message from build.sh or not using it at all

**Comment by leto** [comment]
@tkh23 it is very hard to install different versions of gcc/libc "side-by-side", it might seem to work, just running binaries. But compiling and linking will not work correctly, because different parts of the build will find different versions of gcc/libc/linkers/etc. If you can create a VM with only the new version of gcc, and see if it works, that would be a good test.

**Comment by tkh23** [comment]
Hello Leto and madbuda,
First of all, thanks for your time and concern thus far. Madbuda, I initially tried with the newest version (gcc 7) and no other gcc installed, didn't work. I then tried gcc-6 seeing how you said v7 doesn't work (for anyone wondering, you need to 'export CXX=/usr/bin/gcc-5or6' first and then makepkg from what I've read [here](https://gist.github.com/slizzered/a9dc4e13cb1c7fffec53) - not sure how to confirm it's actually using the older version though).
Leto, I can imagine the complications, however, I didn't build any of the gcc from source but instead got binaries from the official Arch repo. In my experience, they don't put anything there that could conflict, and if so, you will be prompted(forced) to uninstall one before you can install the other. Let's hope they made no mistakes here. Now compiling with gcc-5. Will update.

**Comment by tkh23** [comment]
Expected behavior is to run makepkg -si and end up with an a package that can be installed via pacman -U. 
No success with gcc-5, but different error (about boost).
https://gist.github.com/tkh23/0336e090448a6ced64c4a88a07e7a212
Will run with gcc-6 and gcc-7 next and update this same comment with the logs.
Then I will try to remove all patches Arch applies to zcash from the PKGBUILD. If that fails, I will try to compile manually.

**Comment by tkh23** [comment]
Compile with gcc-7 and Arch's Zcash patches removed from PKGBUILD. Results in different error, althoug boost and libs are installed: 
/usr/bin/ld: cannot find -lboost_system-mt
collect2: error: ld returned 1 exit status
https://gist.github.com/tkh23/95c6763851a2e815c8567b4aec6e7a6f

**Comment by madbuda** [comment]
Looking at your pkgbuild line 54 and 55 should match what we have in build.sh

https://github.com/MyHush/hush/blob/master/zcutil/build.sh#L133

**Comment by tkh23** [comment]
I tried to adjust according to your advice, but I might not have done it quite right. Really wanted to contributed by creating a PKGBUILD, but I think I don't understand the subject matter well enough to continue, so I just built according to your install.md and it does indeed work fine (with gcc 7 btw). Feel free to reach out though with more detailed instructions or requests to test - I have five different arch systems with different CPUs and I'll be glad to help. Thanks!
